zetaclient/maintenance/shutdown_listener.go

```diff
@@ -55,6 +55,38 @@ func (o *ShutdownListener) Listen(ctx context.Context, action func()) {
 	)
 
 	bg.Work(ctx, o.waitForUpdate, bg.WithName("shutdown_listener.wait_for_update"), withLogger, onComplete)
+	bg.Work(ctx, o.checkIfSyncing, bg.WithName("shutdown_listener.check_if_syncing"), withLogger, onComplete)
+}
+
+// checkIfSyncing checks if the node is syncing ,
+// if it is syncing it returns nil which completes the bg task and calls onComplete
+func (o *ShutdownListener) checkIfSyncing(ctx context.Context) error {
```

Comment:
can we call this checkIfNodeSyncing, so it is a bit more obvious? also this is not just checking if its syncing, it is also waiting a bit

Reply:
updated to waitUntilSycing